operators/tokenizer/bpe_utils.hpp

```diff
@@ -359,6 +361,114 @@ class PreTokenizerWithRegEx {
     return {};
   }
 
+  // [^\r\n\p{L}\p{N}]?[\p{Lu}\p{Lt}\p{Lm}\p{Lo}\p{M}]*[\p{Ll}\p{Lm}\p{Lo}\p{M}]+(?i:'s|'t|'re|'ve|'m|'ll|'d)?
+  std::u32string_view Match_PHI4_Pattern_1() {
+    size_t i = 0;
+
+    // [^\r\n\p{L}\p{N}]?
+    if (!IsRN(m_text[i]) && !IsN(m_text[i]) && !IsL(m_text[i])) {
+      i++;
+    }
+
+    // [\p{Lu}\p{Lt}\p{Lm}\p{Lo}\p{M}]*
+    std::vector<ufal::unilib::unicode::category_t> categories1 = {ufal::unilib::unicode::Lu,
+                                                                 ufal::unilib::unicode::Lt,
+                                                                 ufal::unilib::unicode::Lm,
+                                                                 ufal::unilib::unicode::Lo,
+                                                                 ufal::unilib::unicode::M};
+    while (std::find(categories1.begin(), categories1.end(), ufal::unilib::unicode::category(m_text[i])) != categories1.end()){
+      i++;
+    }
+
+    // [\p{Ll}\p{Lm}\p{Lo}\p{M}]+
+    size_t j = i;
+    std::vector<ufal::unilib::unicode::category_t> categories2 = {ufal::unilib::unicode::Ll,
```

Comment:
std::set, and can we make this check to be a IsLl.. function?

Reply:
yeah, figured we wanted to do simply hardcode it for perf since the previous `StringToCategory` and `IsCategory(char32_t ch, std::vector<std::string> categories)` was removed in your last PR; will add them back.

Reply:
Can we just make if/else to compare the type here? the other approach looks like over-design to me.
And extracting them into a function is to dedup the same code since the same sub-pattern appears twice.

Reply:
sounds good!